src/homepageExperience/components/steps/arduino/ExecuteAggregateQuery.tsx

```diff
@@ -0,0 +1,37 @@
+// Libraries
+import React, {useEffect} from 'react'
+
+// Utils
+import {event} from 'src/cloud/utils/reporting'
+import {keyboardCopyTriggered, userSelection} from 'src/utils/crossPlatform'
+
+const logCopyCodeSnippet = () => {
+  event('firstMile.arduinoWizard.executeAggregateQuery.code.copied')
+}
+
+type OwnProps = {
+  bucket: string
+}
+
+export const ExecuteAggregateQuery = (props: OwnProps) => {
+  const {bucket} = props
+
+  useEffect(() => {
+    const fireKeyboardCopyEvent = event => {
+      if (
+        keyboardCopyTriggered(event) &&
+        userSelection().includes('influx query')
+      ) {
+        logCopyCodeSnippet()
+      }
+    }
+    document.addEventListener('keydown', fireKeyboardCopyEvent)
+    return () => document.removeEventListener('keydown', fireKeyboardCopyEvent)
+  }, [])
```

Comment:
This anonymous function in the `useEffect` hook is being repeated in a lot of places. As follow-up work, think of some different implementations that will let you define this function, which creates a keystroke handler function, then assigns it and unassigns it to the DOM, once and then import it where needed. The tricky part will be figuring out how you can the string that the `userSelect().includes` part references.